### PR TITLE
NEW: custom filter fields

### PR DESCRIPTION
**Pull request by unclecheese**
Resolves https://github.com/silverstripe/silverstripe-graphql/issues/406

### CODE REVIEW

**Comment by unclecheese** [comment]
Docs: https://github.com/silverstripe/silverstripe-framework/pull/10192

File: src/Schema/Services/NestedInputBuilder.php [hunk 1]
```diff
@@ -69,13 +69,20 @@ class NestedInputBuilder
...
     /**
      * NestedInputBuilder constructor.
      * @param Field $root
+     * @param Schema $schema
      * @param string $fields
+     * @param array $resolveConfig
      * @throws SchemaBuilderException
      */
-    public function __construct(Field $root, Schema $schema, $fields = Schema::ALL)
+    public function __construct(Field $root, Schema $schema, $fields = Schema::ALL, $resolveConfig = [])
```
Comment (emteknetnz) [inline src/Schema/Services/NestedInputBuilder.php:85]: Add param types e.g. `array $resolveConfig = []`

File: src/Schema/Storage/CodeGenerationStore.php [hunk 2]
```diff
@@ -427,6 +432,17 @@ public function setObfuscator(NameObfuscator $obfuscator): CodeGenerationStore
         return $this;
     }
 
+    /**
+     * If true, s
```
Comment (emteknetnz) [inline src/Schema/Storage/CodeGenerationStore.php:436]: Looks like this is missing some text

File: src/Schema/Services/NestedInputBuilder.php (hunk 1 above)
Comment (emteknetnz) [inline src/Schema/Services/NestedInputBuilder.php:85]: Add param types e.g. `array $resolveConfig = []`

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Looks good

Couple of very minor things

**Pull request merged**